cpp/ql/lib/semmle/code/cpp/security/BufferAccess.qll

```diff
@@ -287,9 +307,10 @@ class FreadBA extends BufferAccess {
 
   override int getSize() {
     result =
-      this.(FunctionCall).getArgument(1).getValue().toInt() *
-        this.(FunctionCall).getArgument(2).getValue().toInt()
+      getSizeExpr().getValue().toInt() * this.(FunctionCall).getArgument(2).getValue().toInt()
```

Comment:
@MathiasVP for cases like this, do you think we should also add a `getSizeMult()` predicate similar to the `AllocationExpr` API? It could be used in this case, and for the other `BufferAccess` subclasses it could return the size of the type of element in the buffer, i.e. `getPointedSize(this.(FunctionCall).getTarget().getParameter(0).getType())` for many of these subclasses.

Reply:
That doesn't sound like a bad idea, no.